scripts/plutus-script-dump.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Run a Cardano local node, and the script dump job
+#
+# Example usage:
+# AWS_ACCESS_KEY_ID=<...> \
+#   AWS_SECRET_ACCESS_KEY=<...> \
+#   AWS_DEFAULT_REGION=<...> \
+#   AWS_ENDPOINT_URL=https://s3.devx.iog.io \
+#   NODE_DIR=../cardano-node-files \
+#   NODE_BIN_DIR=../cardano-node-bin \
+#   S3_DUMP_DIR=s3://plutus/mainnet-script-dump/ \
+#   LOCAL_DUMP_DIR=../mainnet-script-dump \
+#   ./scripts/create-dump.sh
+
+set -eEuo pipefail
+
+trap '(kill $pid_node || true); (kill $pid_dump || true); exit' INT TERM QUIT ERR EXIT
+
+set -x
+
+mkdir -p "$NODE_DIR"
+mkdir -p "$NODE_BIN_DIR"
+mkdir -p "$LOCAL_DUMP_DIR"
+
+# Start running local node
+./scripts/cardano-node.sh > /dev/null 2>&1 &
+
+pid_node=$!
+
+# If the local node is run for the first time, the config file needs to be downloaded,
+# so here we wait until the config file is available.
+while ! [ -f "$NODE_DIR"/mainnet-config.json ]
+do
+  sleep 30
+done
+
+cabal update
+# Start the dump job
+cabal v2-run plutus-script-evaluation-test:dump-script-events -- --socket-path "$NODE_DIR"/db/node.socket --config "$NODE_DIR"/mainnet-config.json --mainnet --blocks-per-file 10000 --events-per-file 50000 --dir "$LOCAL_DUMP_DIR" &
```

Comment:
Huh? This literally won't work in this repo, it works in plutus-apps!

This just makes me more convinced that these scripts should live with the dumper exectuable.

Reply:
Right now it only works in plutus-apps, but once we start using local hackage rather than source-repository-package, I'd move the dumper executable to the plutus repo and make it depend on the streamer in plutus-apps (so that everything related to the test is in the plutus repo), which is why I'm making a PR in this repo.

Reply:
Also the dumper executable currently runs in a non-main branch, because it depends on unmerged changes in cardano-node, and even if they are merged, there's this business of updating cardano-node and other dependencies' commits. So I think we can merge these script here and then copy them to that branch.

Reply:
> (Consider: what happens if plutus master has changes which cause downstream to not compile?)

Fair enough. The dump executable can depend on `plutus-x.y.z` on Hackage rather than the source code, but I can see how one would argue against that :smile: 

I'll move this to `plutus-apps` but this means dealing with two repos and two buildkite pipelines.

Reply:
I do agree this is all a bit ugly but I don't see a better way at the moment :thinking: